src/fondant/dataset/dataset.py

```diff
@@ -508,15 +529,14 @@ def read(
             automatic partitioning
             resources: The resources to assign to the operation.
             cache: Set to False to disable caching, True by default.
+            dataset_name: The name of the dataset.
 
         Returns:
             An intermediate dataset.
         """
-        if self._graph:
-            msg = "For now, at most one read component can be applied per pipeline."
-            raise InvalidPipelineDefinition(
-                msg,
-            )
+        if dataset_name is None:
+            fake = Faker()
+            dataset_name = f"{fake.word()}-{fake.word()}"
```

Comment:
Does this need to be unique? And if it does, could we just use `dataset-{random_number}`? It's a shame that we need a new dependency for this.

Reply:
Convinced - we just need to make sure we have some random dataset name.